sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -208,10 +209,45 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       throw QueryCompilationErrors.pathOptionNotSetCorrectlyWhenReadingError()
     }
 
-    DataSource.lookupDataSourceV2(source, sparkSession.sessionState.conf).flatMap { provider =>
-      DataSourceV2Utils.loadV2Source(sparkSession, provider, userSpecifiedSchema, extraOptions,
-        source, paths: _*)
-    }.getOrElse(loadV1Source(paths: _*))
+    val isUserDefinedDataSource =
```

Comment:
@cloud-fan @allisonwang-db do we want to support this datasource via `USING` syntax unlike DSv2, right?

If that's the case, the logics of loading DataSource has to be within `DataSource.lookupDataSource` and/or `DataSource.providingInstance`. I don't think we should mix the logics here with DSv2.

Reply:
Let's at least separate the logic into a separate function if possible.

Reply:
Unfortunately DS v2 `TableProvider` does not support USING yet. That's why the code is a bit messy here as it's not shared with the SQL USING path. We should support it though...